src/workerd/server/server.c++

```diff
@@ -2205,9 +2205,10 @@ class Server::WorkerService final: public Service,
     }
 
     KJ_IF_SOME(w, workerTracer) {
-      auto tracerSpanObserver =
-          kj::refcounted<WorkerTracerSpanObserver>(kj::refcounted<SpanSubmitter>(kj::addRef(*w)));
-      w->setUserRequestSpan({kj::mv(tracerSpanObserver)});
+      w->setMakeUserRequestSpanFunc([&w = *w]() {
```

Comment:
Non-blocking nit: the capture here *feels* odd. could the reference instead be passed in as an argument when the function is called rather than capturing a bare reference?